include/libtorrent/alert_types.hpp

```diff
@@ -1301,17 +1341,36 @@ namespace libtorrent
 		enum socket_type_t { tcp, tcp_ssl, udp, i2p, socks5, utp_ssl };
 
 		// internal
-		listen_succeeded_alert(aux::stack_allocator& alloc, tcp::endpoint const& ep
+		listen_succeeded_alert(aux::stack_allocator& alloc
+			, libtorrent::address const& listen_addr
```

Comment:
Yes, the compiler just get crazy, not just clang, but gcc too (I saw it in travis). I think that if the field is not called `address` it will be ok, do you want another name?